Sources/NIOHTTP2/HTTP2Settings.swift

```diff
@@ -12,101 +12,232 @@
 //
 //===----------------------------------------------------------------------===//
 import CNIONghttp2
+import NIO
 
-/// A HTTP/2 settings parameter that allows representing both known and unknown HTTP/2
-/// settings parameters.
-public struct HTTP2SettingsParameter {
-    internal let networkRepresentation: UInt16
-
-    /// Create a `HTTP2SettingsParameter` that is not known to NIO.
+/// A single setting for HTTP/2, a combination of parameter identifier and its value.
+public enum HTTP2Setting {
+    /// SETTINGS_HEADER_TABLE_SIZE (0x1)
     ///
-    /// If this is a known parameter, use one of the static values.
-    public init(extensionSetting: Int) {
-        self.networkRepresentation = UInt16(extensionSetting)
-    }
-
-    /// Initialize a `HTTP2SettingsParameter` from nghttp2's representation.
-    internal init(fromNetwork value: Int32) {
-        self.networkRepresentation = UInt16(value)
-    }
-
-    /// A helper to initialize the static parameters.
-    private init(_ value: UInt16) {
-        self.networkRepresentation = value
-    }
-
-    /// Corresponds to SETTINGS_HEADER_TABLE_SIZE
-    public static let headerTableSize = HTTP2SettingsParameter(1)
+    /// Allows the sender to inform the remote endpoint of the maximum size of
+    /// the header compression table used to decode header blocks, in octets. The
+    /// encoder can select any size equal to or less than this value by using
+    /// signaling specific to the header compression format inside a header block.
+    /// The initial value is 4,096 octets.
+    case headerTableSize(Int32)
```

Comment:
Why are these fields `Int32`? They should be `Int` (for programmer convenience).